common/persistence/cassandra/errors.go

```diff
@@ -199,11 +201,27 @@ func extractCurrentWorkflowConflictError(
 
 	actualCurrentRunID := gocql.UUIDToString(record["current_run_id"])
 	if actualCurrentRunID != requestCurrentRunID {
+		binary, _ := record["execution_state"].([]byte)
+		encoding, _ := record["execution_state_encoding"].(string)
+		executionState := &persistencespb.WorkflowExecutionState{}
+		if state, err := serialization.WorkflowExecutionStateFromBlob(binary, encoding); err != nil {
```

Comment:
understand that we need this call here. but we were hoping that we could eliminate encoding/decoding here at this layer? is it reasonable to bubble up the blob from this layer and let upper layer do decoding?

Reply:
>is it reasonable to bubble up the blob from this layer and let upper layer do decoding?

definitely, the serialization / deserialization should not be done at this layer, but for the sake of this internal error migration / deprecation, this need to happen

this PR aims to get rid of this
https://github.com/temporalio/temporal/blob/21aa499c2e9ab0689572a425a52e977b0f0cc2a6/common/persistence/cassandra/cassandraPersistence.go#L937-L1056